src/components/DepositWidget/index.tsx

```diff
@@ -335,6 +335,39 @@ interface BalanceDisplayProps extends TokenLocalState {
   ): Promise<void>
 }
 
+const customFilterFnFactory = (customStopCheck: (...any: unknown[]) => boolean) => (searchTxt: string) => ({
+  symbol,
+  name,
+  address,
+}: TokenBalanceDetails): boolean => {
```

Comment:
Shit, another brain freeze. WTH is going on here?

1. function `customFilterFnFactory` takes as argument a function that returns a boolean.
2. then `customFilterFnFactory` returns a function that:
    1. takes a string as param
    1. then returns a function which:
        1. takes `TokensBalanceDetails` as params
        1. then returns a boolean

Did I get this right?

Reply:
I don't think I got this right. The function body only returns `boolean`s.  :thinking: :question: :confused:

Reply:
you caught sth here actually, it should just take the `localTokens` as the first curried param as it's defined outside the scope. So basically you're right, but it's `localTokens` => `searchText` => `filterFn`